File: FrontEnd_FIT5120/src/services/scamReportService.js
import api from './api.js'

/**
 * 诈骗报告相关API服务
 */
class ScamReportService {
  /**
   * 获取诈骗报告列表（带分页和筛选条件）
   * @param {Object} params - 查询参数
   * @param {number} params.page - 页码，默认1
   * @param {number} params.size - 每页数量，默认10
   * @param {string} params.state - 州/地区筛选（可选）
   * @param {string} params.contactMode - 联系方式筛选（可选）
   * @param {string} params.age - 年龄筛选（可选）
   * @param {string} params.gender - 性别筛选（可选）
   * @param {string} params.categoryLevel2 - 二级分类筛选（可选）
   * @param {string} params.categoryLevel3 - 三级分类筛选（可选）
   * @param {string} params.startDate - 开始日期筛选（可选，格式：yyyy-MM-dd）
   * @param {string} params.endDate - 结束日期筛选（可选，格式：yyyy-MM-dd）
   * @returns {Promise} API响应
   */
  async getScamReports(params = {}) {
    try {
      const response = await api.get('/scam-reports', {
        params: {
          page: params.page || 1,
          size: params.size || 10,
          state: params.state,
          contactMode: params.contactMode,
          age: params.age,
          gender: params.gender,
          categoryLevel2: params.categoryLevel2,
          categoryLevel3: params.categoryLevel3,
          startDate: params.startDate,
          endDate: params.endDate
        }
      })
      return response.data
    } catch (error) {
      console.error('获取诈骗报告失败:', error)
      throw error
    }
  }

  /**
   * 获取诈骗统计数据
   * @returns {Promise} 统计数据
   */
  async getScamStatistics() {
    try {
      const response = await api.get('/scam-reports/statistics')
      return response.data
    } catch (error) {
      console.error('获取诈骗统计数据失败:', error)
      throw error
    }
  }

  /**
   * 获取最新的诈骗警报
   * @param {number} limit - 限制数量，默认5
   * @returns {Promise} 最新警报列表
   */
  async getLatestAlerts(limit = 5) {
    try {
      const response = await api.get('/scam-reports/latest-alerts', {
        params: { limit }
      })
      return response.data
    } catch (error) {
      console.error('获取最新诈骗警报失败:', error)
      throw error
    }
  }
}

export default new ScamReportService()

File: BackEnd_FIT5120/src/main/java/com/example/facilityfinder/service/ScamReportService.java
package com.example.facilityfinder.service;

import com.baomidou.mybatisplus.core.metadata.IPage;
import com.example.facilityfinder.model.ScamReport;

import java.time.LocalDate;
import java.util.List;
import java.util.Map;

/**
 * 诈骗报告服务接口
 */
public interface ScamReportService {

    /**
     * 分页查询诈骗报告（带条件筛选）
     * @param page 当前页
     * @param size 每页大小
     * @param state 州/地址
     * @param contactMode 联系方式
     * @param age 年龄段
     * @param gender 性别
     * @param categoryLevel2 二级分类
     * @param categoryLevel3 三级分类
     * @param startDate 开始日期
     * @param endDate 结束日期
     * @return 分页结果
     */
    IPage<ScamReport> getScamReportsWithConditions(
            Integer page, Integer size,
            String state, String contactMode, String age, String gender,
            String categoryLevel2, String categoryLevel3,
            LocalDate startDate, LocalDate endDate
    );

    /**
     * 获取筛选条件的可选值
     * @return 筛选条件Map
     */
    Map<String, List<String>> getFilterOptions();

    /**
     * 获取统计数据
     * @return 统计数据Map
     */
    Map<String, Object> getStatistics();

    /**
     * 根据ID获取诈骗报告详情
     * @param id 报告ID
     * @return 诈骗报告
     */
    ScamReport getScamReportById(Integer id);

    /**
     * 获取热门诈骗类型（按报告数量排序）
     * @param limit 限制数量
     * @return 热门诈骗类型列表
     */
    List<ScamReport> getPopularScamTypes(Integer limit);
}
